### DISCUSSION TITLE
Open a terminal with current working directory

### DISCUSSION THREAD

**Discussion by MrRostron** [qtile/qtile#5014] [Q&A] [answered]
![2024-09-30_14-15](https://github.com/user-attachments/assets/c46b9d12-6da7-4207-9888-03d1dc0406db)Hey all, so im very new to bash scripts linux etc in general, so i created a simple bash script that can be seen in the top right to spawn a new terminal in the directory im currently working in which works as expected but when i try to run this script from a keybinding it just spawns a terminal in my home directory any help would be appreciated.

**Comment by jrwrigh** [comment]
Issue is that `pwd` will be run from Qtile's process, not the terminal you're currently in. So you'd need to somehow extract the working directory from the terminal (window) currently in focus. That I'm not sure how to do.

PS: In bash and zsh, `$PWD` is already bound and updated to the current working directory. So `PWD=$(pwd)` is technically redundant, even though I personally usually use `pwd` rather than `$PWD` myself.

**Reply by MrRostron** [reply]
thank you for your reply i found a solution here https://github.com/alacritty/alacritty/issues/808#issuecomment-334200570

**Comment by MrRostron** [comment, marked as answer]
https://github.com/alacritty/alacritty/issues/808#issuecomment-334200570
